lib/Service/UserConfigService.php

```diff
@@ -51,17 +51,21 @@ public function __construct(
 		$this->rootFolder = $rootFolder;
 	}
 
-	public function getUserConfig(string $key) {
+	public function getUserConfig(string $key): string {
+		$user = $this->userSession->getUser();
+		return $this->getConfigForUser($user->getUid(), $key);
+	}
+
+	public function getConfigForUser(string $userId, string $key): string {
 		if (!in_array($key, array_keys(self::DEFAULT_CONFIGS))) {
 			throw new Exception('Unknown user config key');
 		}
-		$user = $this->userSession->getUser();
 		$default = self::DEFAULT_CONFIGS[$key];
-		$value = $this->config->getUserValue($user->getUid(), Application::APP_ID, $key, $default);
+		$value = $this->config->getUserValue($userId, Application::APP_ID, $key, $default);
 
 		// If the config is a path, make sure it exists
 		if (str_starts_with($default, '/')) {
-			$userFolder = $this->rootFolder->getUserFolder($user->getUID());
+			$userFolder = $this->rootFolder->getUserFolder($userId);
 			// If the folder does not exists, create it
 			if (!$userFolder->nodeExists($value)) {
 				$userFolder->newFolder($value);
```

Comment:
Move Folder creatin logic to a protected in a protected `AlbumRoot`.
So we can call it straight from `AlbumRoot->createFile` as well from `PublicAlbumRoot` and  `SharedAlbumRoot`